src/main/java/stream/flarebot/flarebot/FlareBot.java

```diff
@@ -188,25 +188,97 @@ public void onError(SimpleLog log, Throwable err) {
         });
         Spark.port(8080);
 
-        CommandLineArguments.parse(args);
+        JSONConfig config = null;
+        try {
+            File file = new File("config.json");
+            if (!file.exists())
+                file.createNewFile();
+            try {
+                config = new JSONConfig("config.json");
+            } catch (NullPointerException e) {
+                FlareBot.LOGGER.error("Invalid JSON!", e);
+                System.exit(1);
+            }
+        } catch (IOException e) {
+            FlareBot.LOGGER.error("Unable to create config.json!", e);
+            System.exit(1);
+        }
+
+        List<String> required = new ArrayList<>();
+        required.add("bot.token");
+        required.add("bot.statusHook");
+        required.add("cassandra.username");
+        required.add("cassandra.password");
+        required.add("mysc.yt");
+
+        boolean good = true;
+        for (String req : required) {
+            if (config.getString(req) != null) {
+                if (!config.getString(req).isPresent()) {
+                    good = false;
+                    LOGGER.error("Missing required json " + req);
+                }
+            } else {
+                good = false;
+                LOGGER.error("Missing required json " + req);
+            }
+        }
 
-        String tkn = CommandLineArguments.TOKEN.getValue();
-        passwd = CommandLineArguments.SQL_PW.getValue();
+        if (!good) {
+            LOGGER.error("One or more of the required JSON objects where missing. Exiting to prevent problems");
+            System.exit(1);
+        }
 
-        new CassandraController().init();
+        String tkn = config.getString("bot.token").get();
+
+        FlareBot.user = config.getString("cassandra.username").get();
+        FlareBot.pass = config.getString("cassandra.password").get();
 
-        FlareBot.secret = CommandLineArguments.SECRET.getValue();
-        FlareBot.dBotsAuth = CommandLineArguments.DBOTS.getValue();
-        FlareBot.webSecret = CommandLineArguments.WEBSECRET.getValue();
-        if (CommandLineArguments.DEBUG.isSet()) {
-            ((ch.qos.logback.classic.Logger) LoggerFactory
-                    .getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME))
-                    .setLevel(Level.DEBUG);
+        if (config.getArray("cassandra.nodes").isPresent()) {
```

Comment:
You shouldn't handle Cassandra here, you can use the config in the CassandraController init. There is no need to use it here and do this. You are just making variables that will be used once and just exposing more info than is needed.

Reply:
so should i open the config to everything else then?